packages/toolpad-components/src/Chart.tsx

```diff
@@ -0,0 +1,217 @@
+import * as React from 'react';
+import { createComponent } from '@mui/toolpad-core';
+import { Container, ContainerProps } from '@mui/material';
+import {
+  XAxis,
+  YAxis,
+  Tooltip,
+  Legend,
+  Line,
+  ResponsiveContainer,
+  CartesianGrid,
+  ComposedChart,
+  Bar,
+  Area,
+  Scatter,
+} from 'recharts';
+import { uniq } from 'lodash-es';
+import { SX_PROP_HELPER_TEXT } from './constants';
+
+export const CHART_DATA_SERIES_KINDS = ['line', 'bar', 'area', 'scatter'];
+
+export interface ChartDataSeries<D = Record<string, string | number>> {
+  kind: (typeof CHART_DATA_SERIES_KINDS)[number];
+  label: string;
+  data?: D[];
+  xKey: keyof D;
+  yKey: keyof D;
+  color?: string;
+}
+
+export type ChartData = ChartDataSeries[];
+
+function getBarChartDataSeriesNormalizedYKey(dataSeries: ChartDataSeries, index: number): string {
+  return `${dataSeries.label}-${dataSeries.yKey}-${index}`;
+}
+
+interface ChartProps extends ContainerProps {
+  data?: ChartData;
+  height?: number;
+}
+
+function Chart({ data = [], height, sx }: ChartProps) {
+  const xValues = React.useMemo(
+    () =>
+      uniq(
+        data
+          .map((dataSeries) =>
+            (dataSeries.data || []).map((dataSeriesPoint) => dataSeriesPoint[dataSeries.xKey]),
+          )
+          .flat()
```

Comment:
Why does this need to be flattened? Bad user input? Maybe it's better to make sure they're aware of this instead of guessing that this is what they intended?

Reply:
I'm flattening to have all the unique x values from every data series into a single array, as the x-axis is shared between all charts and it helps to have it for some things such as setting the domain of the x-axis that all the charts share.
If you see this affecting the user experience in any particular scenario I can look into different approaches.

Reply:
Ah, right I misunderstood. You can also just `.flatMap` and avoid creating an array of arrays altogether